definitions/npm/ramda_v0.26.x/flow_v0.104.x-/test_ramda_v0.26.x_list.js

```diff
@@ -822,4 +818,60 @@ const str: string = "hello world";
     s: a,
     y: b
   }))([1, 2, 3])(["1", "2", "3"]);
+
+  describe('nth', () => {
+    it('should get a maybe number back on input is an array of numbers', () => {
+      const nthxs1: ?number = nth(2, [1,2,3]);
+      const nthxs2: ?number = nth(2)([1,2]);
+    });
+
+    it('should get a maybe string back on input is an arary of strings', () => {
+      const nthxs3: ?string = nth(2, ["curry"]);
+      const nthxs4: ?string = nth(2)(["curry", "bean"]);
+    });
+
+    it('should get a string back when input is string', () => {
+      const nthxs5: string = nth(2)("curry");
+      const nthxs6: string = nth(2, "curry");
+    });
+
+    it('should take read only array as second argument', () => {
+      const readOnlyArray: $ReadOnlyArray<number>  = [1, 2, 3];
+      const nthxs = nth(2)(readOnlyArray)
+      const nthxs1 = nth(2, readOnlyArray)
+    });
+
+    it('should take a mutable array as second arugment', () => {
+      const array: Array<string> = ['foo', 'bar'];
+      const nthxs = nth(1)(array)
+      const nthxs1 = nth(1, array)
+    });
+
+    it('should fail on expecting return type to be a maybe string', () => {
+      //$ExpectError
+      const nthxs: ?string = nth(2)([1, 2, 3]);
+    });
+
+    it('should fail on expecting return type to be string when input array consists of numbers', () => {
```

Comment:
Sometimes we just write another set of tests for varying arities, but given that the arities here are known, I kind of like this approach of putting the same assertions into a single test. Good job!